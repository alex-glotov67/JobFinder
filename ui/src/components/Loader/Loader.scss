@import "../../styles/variables";

.loader {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100%;
  z-index: 1;
  height: 100%;
  border-radius: 0;
  transform: translateY(-50%) translateX(-50%);
  background: $transparent-loader;

  &.initial {
    background: $white;
  }

  &-circle {
    width: 4.375rem;
    height: 4.375rem;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    background-color: transparent;
    border-radius: 50%;
    border: 0.1875rem solid $main-color;
    animation: preloader-animation 0.7s infinite alternate;

    &:nth-child(2) {
      width: 2.5rem;
      height: 2.5rem;
      animation-delay: 0.05s;
    }
    &:nth-child(3) {
      width: 1.25rem;
      height: 1.25rem;
      animation-delay: 0.1s;
    }
  }
}

@keyframes preloader-animation {
  0% { transform: translateX(-50%) translateY(-50%) scale(0.5) }
  100% { transform: translateX(-50%) translateY(-50%) scale(1) }
}