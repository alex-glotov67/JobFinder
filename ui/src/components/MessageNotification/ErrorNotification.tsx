import React from 'react';

interface ErrorNotificationProps {
  errors: string | undefined;
}

const ErrorNotification = ({ errors }: ErrorNotificationProps) => (
  <>{errors && <div className="input-field--errors">{errors}</div>}</>
);

export default ErrorNotification;
