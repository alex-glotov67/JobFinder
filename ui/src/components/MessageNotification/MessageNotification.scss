@import "../../styles/variables";

.message {
  padding: .75rem 1rem;
  text-align: center;
  margin-bottom: 0.8rem;
  &-error {
    color: map-get($error-message, text-color);
    background-color: map-get($error-message, bg-color);
    border-color: map-get($error-message, bd-color);
  }
  &-success {
    color: map-get($success-message, text-color);
    background-color: map-get($success-message, bg-color);
    border-color: map-get($success-message, bd-color);
  }
}