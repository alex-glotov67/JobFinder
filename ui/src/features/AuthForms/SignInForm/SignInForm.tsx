import React from 'react';
import { useFormik } from 'formik';
import { useDispatch } from 'react-redux';

import InputField from '../../../components/Input/InputField';
import Icon from '../../../components/Icon/Icon';
import InputPasswordField from '../../../components/Input/InputPasswordField';
import { Button } from '../../../components/Button/Button';
import { AuthFormComponent } from '../../../types/interfaces/AuthFormComponent';
import { signInLoadAction } from '../../../redux/actions/SignInActions';
import { signInSchema } from '../../../utils/validation/signInSchema';

const SignInForm = () => {
  const dispatch = useDispatch();
  const initialValues: AuthFormComponent = {
    email: '',
    password: '',
  };
  const { handleSubmit, handleChange, values, errors } = useFormik({
    initialValues,
    validateOnChange: false,
    validateOnBlur: false,
    enableReinitialize: true,
    validationSchema: signInSchema,
    onSubmit(values) {
      dispatch(signInLoadAction(values));
    },
  });
  return (
    <>
      <form onSubmit={handleSubmit}>
        <InputField
          type="text"
          placeholder="Enter an email"
          name="email"
          onChange={handleChange}
          value={values.email}
          errors={errors.email}
        >
          <Icon iconName="icon-mail" />
        </InputField>
        <InputPasswordField
          type="password"
          placeholder="Enter a password"
          name="password"
          onChange={handleChange}
          value={values.password}
          errors={errors.password}
        />
        <div className="auth__inner-buttons">
          <Button type="submit">Sign in</Button>
        </div>
      </form>
    </>
  );
};

export default SignInForm;
