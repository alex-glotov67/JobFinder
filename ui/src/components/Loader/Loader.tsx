import React from 'react';
import classNames from 'classnames';

import './Loader.scss';

interface LoaderProps {
  init?: boolean;
}

export const Loader = ({ init }: LoaderProps) => (
  <div className={classNames('loader', init && 'initial')}>
    {[...Array(3).keys()].map((item, index) => (
      <div key={index + item} className="loader-circle" />
    ))}
  </div>
);
