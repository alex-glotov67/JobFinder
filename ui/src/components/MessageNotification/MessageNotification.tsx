import React from 'react';
import classNames from 'classnames';

import './MessageNotification.scss';

interface MessageNotificationProps {
  error?: string;
  success?: string;
}

export const MessageNotification = ({
  error,
  success,
}: MessageNotificationProps) => {
  return (
    <div
      className={classNames(
        'message',
        error ? 'message-error' : 'message-success',
      )}
    >
      {error || success}
    </div>
  );
};
