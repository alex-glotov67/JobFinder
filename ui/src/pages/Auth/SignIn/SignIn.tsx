import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import SignInForm from '../../../features/AuthForms/SignInForm/SignInForm';
import { signUpResetAction } from '../../../redux/actions/SignUpActions';
import {
  getSignInSelector,
  getSignUpSelector,
} from '../../../redux/selectors/authSelector';
import { Loader } from '../../../components/Loader/Loader';
import { MessageNotification } from '../../../components/MessageNotification/MessageNotification';

import '../Auth.scss';

const SignIn = () => {
  const dispatch = useDispatch();
  const { errorMessage, isLoading } = useSelector(getSignInSelector);
  const signUpSelector = useSelector(getSignUpSelector);

  useEffect(() => {
    return () => {
      dispatch(signUpResetAction());
    };
  });

  return (
    <div className="auth">
      <div className="container">
        <div className="auth__inner">
          <div className="auth__inner-card">
            <div className="auth__inner-card--content">
              <h1 className="auth__inner-title">Sign in</h1>
              {errorMessage ? (
                <MessageNotification error={errorMessage?.error} />
              ) : signUpSelector.isFetching && !errorMessage ? (
                <MessageNotification
                  success={'An account was successfully registered'}
                />
              ) : (
                ''
              )}
              <SignInForm />
              {isLoading && <Loader />}
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default SignIn;
