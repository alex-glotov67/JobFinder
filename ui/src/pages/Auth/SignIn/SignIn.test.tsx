import React from 'react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SignIn from './SignIn';
import { MessageNotification } from '../../../shared/MessageNotification/MessageNotification';
import SignInForm from '../../../components/AuthForms/SignInForm/SignInForm';
import { Loader } from '../../../shared/Loader/Loader';
const mockStore = configureMockStore();

describe('SignIn Page', () => {
  it('should render sign in form', () => {
    const store = mockStore({
      signInReducer: {
        errorMessage: null,
      },
      signUpReducer: {
        isFetching: false,
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <SignIn />
      </Provider>,
    );
    expect(wrapper.contains(<SignInForm />)).toBe(true);
  });
  it('should render loading', () => {
    const store = mockStore({
      signInReducer: {
        isLoading: true,
      },
      signUpReducer: {
        isFetching: false,
      },
    });
    const wrapper = mount(
      <Provider store={store}>
        <SignIn />
      </Provider>,
    );
    expect(wrapper.contains(<Loader />)).toBe(true);
  });
  describe('Error testing', () => {
    it('should be rendered with error', () => {
      const store = mockStore({
        signInReducer: {
          errorMessage: {
            error: 'Test error',
          },
        },
        signUpReducer: {
          isFetching: false,
        },
      });
      const wrapper = mount(
        <Provider store={store}>
          <SignIn />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
      const errorComponent = wrapper.find(MessageNotification);
      expect(errorComponent.props().error).toEqual('Test error');
    });
    it('should be rendered with success message', () => {
      const store = mockStore({
        signInReducer: {
          errorMessage: null,
        },
        signUpReducer: {
          isFetching: true,
        },
      });
      const wrapper = mount(
        <Provider store={store}>
          <SignIn />
        </Provider>,
      );
      const successComponent = wrapper.find(MessageNotification);
      expect(successComponent.props().success).toEqual(
        'An account was successfully registered',
      );
    });
  });
});
