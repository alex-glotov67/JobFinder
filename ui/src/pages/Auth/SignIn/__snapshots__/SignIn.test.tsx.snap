// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Page Error testing should be rendered with error 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SignIn>
    <div
      className="auth"
    >
      <div
        className="container"
      >
        <div
          className="auth__inner"
        >
          <div
            className="auth__inner-card"
          >
            <div
              className="auth__inner-card--content"
            >
              <h1
                className="auth__inner-title"
              >
                Sign in
              </h1>
              <Component
                error="Test error"
              >
                <div
                  className="message message-error"
                >
                  Test error
                </div>
              </Component>
              <SignInForm>
                <form
                  onSubmit={[Function]}
                >
                  <InputField
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter an email"
                    type="text"
                    value=""
                  >
                    <div
                      className="input-field input-field--icon"
                    >
                      <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Enter an email"
                        type="text"
                        value=""
                      />
                      <Icon
                        iconName="icon-mail"
                      >
                        <span
                          className="icon icon-mail"
                        />
                      </Icon>
                    </div>
                  </InputField>
                  <InputPasswordField
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter a password"
                    type="password"
                    value=""
                  >
                    <div
                      className="input-field input-field--password"
                    >
                      <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Enter a password"
                        type="password"
                        value=""
                      />
                      <Icon
                        iconName="icon-eye"
                        onClick={[Function]}
                      >
                        <span
                          className="icon icon-eye"
                          onClick={[Function]}
                        />
                      </Icon>
                      <div
                        className="input-field--errors"
                      />
                    </div>
                  </InputPasswordField>
                  <div
                    className="auth__inner-buttons"
                  >
                    <Component
                      type="submit"
                    >
                      <button
                        className="btn primary"
                        type="submit"
                      >
                        Sign in
                      </button>
                    </Component>
                  </div>
                </form>
              </SignInForm>
            </div>
          </div>
        </div>
      </div>
    </div>
  </SignIn>
</Provider>
`;
